blockly-core/core/utils/xml.js

```diff
@@ -362,6 +369,11 @@ Blockly.Xml.domToBlock = function(blockSpace, xmlBlock) {
   if (userCreated) {
     block.userCreated = (userCreated === 'true');
   }
+  var limit = xmlBlock.getAttribute('limit');
+  var shouldShowLimits = blockSpace.isFlyout || Blockly.editBlocks;
```

Comment:
Should this be an xor [as expressed here](https://github.com/code-dot-org/code-dot-org/pull/8751/files#diff-19bada970605b4e1a316b8460d76be00R203)?